python/modules/BtagSFWeightProducer.py

```diff
@@ -0,0 +1,119 @@
+import ROOT
+import os
+import numpy as np
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+
+class BtagSFWeightProducer(Module):
+
+    def __init__(self, bTagEffFile, sampleName, bDiscCut, jetPtMin = 20, jetEtaMax = 2.4, fileDirectory = os.environ['CMSSW_BASE'] + "/src/PhysicsTools/NanoSUSYTools/data/btagSF/"):
+        self.jetPtMin = jetPtMin
+        self.jetEtaMax = jetEtaMax
+        self.bDiscCut = bDiscCut
+
+        self.bTagEffFile = bTagEffFile
+        self.sampleName = sampleName
+        self.fileDirectory = fileDirectory
+
+    def beginJob(self):
+        ROOT.TH1.AddDirectory(False)
+        
+        fin = ROOT.TFile.Open(self.fileDirectory + "/" + self.bTagEffFile)
+
+        self.h_eff_b          = fin.Get(("n_eff_b_" + self.sampleName));
```

Comment:
You might run into samples without the pregenerated n_eff in the root file. Is there a fallback or error control?

Reply:
Thanks for the reminder, I had meant to put in such a safety, but obviously didn't.

Reply:
I pushed another commit to address this.